nodebuilder/store_test.go

```diff
@@ -160,6 +162,103 @@ func TestStoreRestart(t *testing.T) {
 	}
 }
 
+func TestDiscoverOpened(t *testing.T) {
+	t.Run("single open store", func(t *testing.T) {
+		_, dir := initAndOpenStore(t, node.Full)
+
+		mockDefaultNodeStorePath := func(t string, n string) (string, error) {
+			return dir, nil
+		}
+		DefaultNodeStorePath = mockDefaultNodeStorePath
+
+		path, err := DiscoverOpened()
+		require.NoError(t, err)
+		require.Equal(t, dir, path)
+	})
+
+	t.Run("multiple open nodes by preference order", func(t *testing.T) {
+		networks := []p2p.Network{p2p.Mainnet, p2p.Mocha, p2p.Arabica, p2p.Private}
+		nodeTypes := []node.Type{node.Bridge, node.Full, node.Light}
+
+		// Store opened stores in a map (network + node -> dir/store)
+		dirMap := make(map[string]string)
+		storeMap := make(map[string]Store)
+		for _, network := range networks {
+			for _, tp := range nodeTypes {
+				store, dir := initAndOpenStore(t, tp)
+				key := network.String() + "_" + tp.String()
+				dirMap[key] = dir
+				storeMap[key] = store
+			}
+		}
+
+		mockDefaultNodeStorePath := func(tp string, n string) (string, error) {
+			key := n + "_" + tp
+			return dirMap[key], nil
+		}
+		DefaultNodeStorePath = mockDefaultNodeStorePath
```

Comment:
Interesting testing approach with replacing global func. Should mockDefaultNodeStorePath return some error in case when key is not present in the map, as we expect it to be there at that point?

Reply:
yeah, figured this was simplest.. I added in the error return but shouldn't impact the test in its current form